### PR TITLE
update for add_folder and set_item_order stuff

### PR DESCRIPTION
**Pull request by markhornak**
### Overview

Shortcut: [111708](https://app.shortcut.com/guru/story/111708/update-to-folder-data-object)

Small'ish updates:
-  to add a method to the Folder class, `add_folder` that provides a shortcut to adding a folder to the current folder.
-  to add `set_folder_item_order` that provides a way for a user to re-order the items in a folder by providing  an ordered list of item names.
- update to `publish_folders.py` to update the final call from `publish_boards` to `publish_folder`.  oversight in original update

### CODE REVIEW

File: guru/data_objects.py [hunk 1]
```diff
@@ -102,6 +102,10 @@ def __init__(self, data, folder_items=[], guru=None, parent_folder=None):
     self.__cards = []
     self.__folders = []
 
+    # special array for internal sorting of items,
+    # cliff notes: needed b/c it is referenced in core.py and can't be an __variable due to mangling.
+    self.sortedItems = []
```
Comment (jahlbornG) [inline guru/data_objects.py:107]: this doesn't make any sense to me.  why wouldn't the caller just use `items()` to get the items after they have been sorted?

File: guru/data_objects.py [hunk 2]
```diff
@@ -324,6 +337,45 @@ def move_to_collection(self, collection, timeout=0):
...
+            boolean : True if it worked, False if there was an error
+    """
+    def get_key(b):
+      for i in range(len(items)):
+        if b.title.lower().strip() == items[i].lower().strip():
+          return i
+      # if we couldn't find it, move it to the back.
+      return len(items)
+
+    # sort items based on the keys passed in.
```
Comment (jahlbornG) [inline guru/data_objects.py:370]: need to check that `__items` is initialized

File: guru/data_objects.py (hunk 1 above)
Comment (markhornak) [reply guru/data_objects.py:107]: updated this to use `self.__items` for the sort.

File: guru/data_objects.py (hunk 2 above)
Comment (markhornak) [reply guru/data_objects.py:374]: added check for `self.__has_items()` and will call `get_items()` if false

File: guru/data_objects.py (hunk 1 above)
Comment (jahlbornG) [inline guru/data_objects.py:107]: nit, don't need this anymore, right?

**Review by jahlbornG — APPROVED** [review]

**Pull request closed without merging**